packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx

```diff
@@ -0,0 +1,209 @@
+ *
+ * @slot default - The required slotted facet.
+ * @part popover-button - The button to click to display or hide the popover menu.
+ * @part label - The associated facet label.
+ * @part value-count - Number of selected values for the facet
+ * @part arrow-icon - The arrow icon on the button to display or hide the popover menu.
+ * @part placeholder - The placeholder displayed when the facet is loading.
+ * @part slot-wrapper - The wrapper that contains the 'facet' or 'slot'.
+ */
+@Component({
+  tag: 'atomic-popover',
+  styleUrl: 'atomic-popover.pcss',
+  shadow: true,
+})
+export class AtomicPopover implements InitializableComponent {
+  @Element() host!: HTMLElement;
+  private hasValues?: () => boolean;
+  private numberOfSelectedValues?: () => number;
+  @InitializeBindings()
+  public bindings!: Bindings;
+  private searchStatus!: SearchStatus;
+
+  @BindStateToController('searchStatus')
+  @State()
+  public searchStatusState!: SearchStatusState;
+  @BindStateToController('facet')
+  @State()
+  public facetState!: FacetState;
+  @State()
+  public error!: Error;
+  @State() private isOpen = false;
+  @State() private facetInfo?: FacetInfo;
+
+  @Event({
+    eventName: 'atomic/closePopovers',
+  })
+  private closePopovers!: EventEmitter<ClearPopoverEvent>;
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+
+    if (this.host.children.length === 0) {
+      this.error = new Error(
+        'One child is required inside a set of popover tags.'
+      );
+
+      return;
+    }
+
+    if (this.host.children.length > 1) {
+      this.error = new Error(
+        'Cannot have more than one child inside a set of popover tags.'
+      );
+    }
+  }
+
+  @Listen('atomic/closePopovers', {target: 'document'})
+  public popOpened(event: CustomEvent<ClearPopoverEvent>) {
+    if (event.detail.ownId !== this.facetInfo?.facetId) {
+      this.isOpen = false;
+    }
+  }
+
+  @Listen('atomic/initializePopover')
+  public initializePopover(event: CustomEvent<InitializePopoverEvent>) {
+    if (this.facetInfo || !event.detail) {
+      return;
+    }
+
+    this.facetInfo = event.detail.facetInfo;
+    this.facetInfo.element.classList.add(popoverClass);
+    this.hasValues = event.detail.getHasValues;
+    this.numberOfSelectedValues = event.detail.getNumberOfSelectedValues;
+  }
+
+  private get popoverId() {
+    return `${this.facetInfo?.facetId}-popover`;
+  }
+
+  private togglePopover() {
+    if (!this.isOpen) {
+      this.closePopovers.emit({ownId: this.facetInfo?.facetId});
+    }
+
+    this.isOpen = !this.isOpen;
+  }
+
+  private renderDropdownButton() {
+    const label = this.bindings.i18n.t(this.facetInfo!.label);
+    const hasSelectedValues = !!this.numberOfSelectedValues?.();
+    const count = this.numberOfSelectedValues?.().toLocaleString();
+    const ariaLabel = this.bindings.i18n.t(
+      this.isOpen ? 'close-popover' : 'open-popover',
+      {label}
+    );
```

Comment:
I would avoid toggling between two different labels since `aria-expanded` should already be conveying this (at-least in theory, we should probably test with screen readers). I would use a label like "Popup filter for {{label}}".